core/src/main/java/org/hibernate/ogm/util/impl/ArrayHelper.java

```diff
@@ -22,7 +23,13 @@
 	}
 
 	public static String[][] to2DStringArray(Collection coll) {
-		return (String[][]) coll.toArray( new String[ coll.size() ][] );
+                String[][] arr2D = new String[coll.size()][];
+                Iterator it = coll.iterator();
+                int i = 0;
+                for (;it.hasNext();i++) {
+                    arr2D[i] = new String[]{ (String) it.next()};
+                }                
+		return arr2D;
```

Comment:
Because
________________
java.lang.ArrayStoreException: null
        at java.lang.System.arraycopy(Native Method)
        at java.util.Arrays$ArrayList.toArray(Arrays.java:3833)
        at org.hibernate.ogm.util.impl.ArrayHelper.to2DStringArray(ArrayHelper.java:33)
        at org.hibernate.ogm.util.impl.ArrayHelperTest.testTo2DStringArray(ArrayHelperTest.java:35)